interop-testing/src/main/java/io/grpc/testing/integration/TestServiceClient.java

```diff
@@ -519,30 +525,33 @@ private void runTest(TestCases testCase) throws Exception {
       case RPC_SOAK: {
         tester.performSoakTest(
             serverHost,
-            false /* resetChannelPerIteration */,
             soakIterations,
             soakMaxFailures,
             soakPerIterationMaxAcceptableLatencyMs,
             soakMinTimeMsBetweenRpcs,
             soakOverallTimeoutSeconds,
             soakRequestSize,
-            soakResponseSize);
+            soakResponseSize,
+            numThreads,
+            (currentChannel) -> tester.maybeCreateNewChannel(currentChannel, false)
```

Comment:
nit: we might actually make this more readable if we removed the boolean parameter from `tester.maybeCreateNewChannel`

I.e.:

1) rename `tester.maybeCreateNewChannel` to `tester.createNewChannel`

2) In rpc_soak cases, just pass a lambda like this: `(currentChannel) -> currentChannel`

3) In channel_soak cases, pass `(currentChannel) -> tester.createNewChannel(currentChannel)`